src/arguments.cpp

```diff
@@ -212,6 +213,46 @@ std::tuple<IpVer, std::string, uint8_t> Arguments::asIpAddrMask()
     throw std::invalid_argument(err);
 }
 
+std::string Arguments::asIpOrFQDN()
+{
+    const char* arg = asText();
+
+    try
+    {
+        auto [_, addr] = parseIpAddress(arg);
+        return addr;
+    }
+    catch (const std::invalid_argument&)
+    {
+        // pass
+    }
+
+    static const std::regex r(
+        // A full domain name is limited to 255 octets (
+        // (including separators)
+        "(?=^.{1,255}$)"
+        "(^"
+        // - The length of any one label is limited to between 1 and 63
+        //   octets.
+        // - labels must not start or end with hyphens.
+        // - Total number of lables is limited to 127
+        "((?!-)[a-z0-9-]{0,62}[a-z0-9]\\.){0,126}"
+        // Trailing dot is optional
+        "((?!-)[a-z0-9-]{0,62}[a-z0-9]\\.?)"
+        "$)",
+        std::regex_constants::icase);
+
+    if (std::regex_match(arg, r))
+    {
+        return arg;
+    }
+
+    std::string err = "Invalid argument: ";
+    err += arg;
+    err += ", expected IP address or FQDN hostname";
```

Comment:
`FQDN hostname` sounds strange. FQDN is already "fully-qualified domain name", hostname is just the first label of it. So better say just `IP address or FQDN`.

Reply:
done